source/PlayerInfo.cpp

```diff
@@ -2354,7 +2364,40 @@ void PlayerInfo::StepMissions(UI *ui)
 		else if(mission.CanComplete(*this))
 			RemoveMission(Mission::COMPLETE, mission, ui);
 		else if(mission.Destination() == GetPlanet() && !freshlyLoaded)
+		{
 			mission.Do(Mission::VISIT, *this, ui);
+			if(mission.IsUnique())
+				continue;
+			
+			// On visit dialogs are handled separately as to avoid a player
+			// getting spammed by on visit dialogs if they are stacking jobs
+			// from the same destination.
+			auto action = mission.GetAction(Mission::VISIT);
+			if(!action.DialogText().empty())
+			{
+				if(visitText.empty())
+					visitText = Format::Replace(ait.DialogText(), substitutions);
+				else
+					++extraVisitDialogs;
+				if(mission.Deadline())
+					++deadlineMissions;
+			}
+		}
+	}
+	if(!visitText.empty())
+	{
+		if(extraVisitDialogs)
```

Comment:
This PR currently only counts "other unfinished missions" as those that attempted to display a visit dialog. However, it is not a strict requirement that a mission must have a visit dialog, so it is possible for the player to have a great number of pips for a given planet, and yet be told "you have only 1 other unfinished mission". Thoughts?

Reply:
If none of the unfinished missions have visit dialogs, would something still pop up? Or would it only be when there is a visit dialog that it also tells you however many other missions you have unfinished?

Reply:
I think, if we are telling the player a number of unfinished missions for their port, that the number needs to be accurate, regardless of whether the missions have a visit message. We could change the phrasing too, and mention something like "you received N other, similar messages"

Reply:
So make it a default feature that you'll be told about unfinished missions?

Reply:
Or change the message text. I don't want the pop-up to say something like "you have 1 other unfinished mission here" when the player can see on the map that they have at least 12 unfinished missions there.

Reply:
Or change the message text. I don't want the pop-up to say something like "you have 1 other unfinished mission here" when the player can see on the map that they have at least 12 unfinished missions there.

Reply:
I've changed it so that all unfinished missions are counted, but it only tells you that if any one of those unfinished missions has an on visit dialog.